.github/workflows/windows-macos.yml

```diff
@@ -30,17 +30,59 @@ jobs:
     steps:
     - name: Determine build matrix
       id: set-matrix
-      run: |
-        set -euo pipefail
-
-        MATRIX='{"include": ['
-        MATRIX+='{"runs-on": "windows-latest"},'
-        MATRIX+='{"runs-on": "macos-13", "arch": "x86_64", "mono": "mono"},'
-        MATRIX+='{"runs-on": "macos-14", "arch": "arm64"},'
-        MATRIX+='{"runs-on": "macos-latest", "arch": "arm64"}'
-        MATRIX+=']}'
-
-        echo "matrix=${MATRIX}" >> $GITHUB_OUTPUT
+      uses: actions/github-script@v7
+      with:
+        script: |
+          const matrix = { include: [] };
+
+          // Windows targets
+          matrix.include.push({
+              "name": "Windows Server 2025",
+              "runs-on": "windows-2025"
+          });
+
+          // macOS targets
+          matrix.include.push({
+            "name": "macOS 13 Ventura (Intel Silicon)",
+            "runs-on": "macos-13",
+          },{
+            "name": "macOS 14 Sonoma (Apple Silicon)",
+            "runs-on": "macos-14",
+          },{
+            "name": "macOS Latest (Apple Silicon)",
+            "runs-on": "macos-latest",
+            // ##############################################################
+            // "nuget" and "mono" are removed from the macos-15 images, see:
+            // https://github.com/actions/runner-images/issues/10686
+            // ##############################################################
+            "brew_extra_pkgs": ["nuget", "mono"]
+          },{
+            "name": "macOS 15 Sequoia (Intel Silicon)",
+            "runs-on": "macos-15-intel",
+            // ##############################################################
+            // "nuget" and "mono" are removed from the macos-15 images, see:
+            // https://github.com/actions/runner-images/issues/10686
+            // ##############################################################
+            "brew_extra_pkgs": ["nuget", "mono"],
+            "optional": true
+          })
+
+          // Auto populate fields
+          for (const job of matrix.include) {
+            if (job["runs-on"]?.startsWith("macos"))
+              job.mono = "mono"
+
+            switch (true) {
+              case job["runs-on"]?.endsWith("-intel"):
+                job.arch = "x86_64";
+                break;
+              case job["runs-on"]?.endsWith("-arm"):
+                job.arch = "arm64";
+                break;
+            }
```

Comment:
I think I would rather just manually add this into the above dictionary literal.

Reply:
Yes, I initially used the name field for this, but the name ended up being changed to a more appealing one. This is a bit overkill and redundant at this point, too.

Reply:
The weird thing is that the CI passes :shrug: Could that mean the `arch` is redundant at this point? I'll take a look.

Reply:
Found the original commit:

https://github.com/canonical/multipass/commit/bbe964b893df660ca52f2a7acf68914405068c2f

So, it seems like GH ARM runners were not a thing back then, so the arch wrapper is basically needed for cross-compiling macOS packages for the different target arch. That's a thing we're no longer doing since the arch-specific packages are being built in their native host environment (i.e., Intel and ARM silicon, respectively).

I'll move the arch vars to each respective JSON node, and submit another PR to remove the arch wrapper altogether.